ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/TimeModel.scala

```diff
@@ -34,7 +34,7 @@ case class TimeModel private (
     val lowerBound = minRecordTime(ledgerTime)
     val upperBound = maxRecordTime(ledgerTime)
     if (recordTime.isBefore(lowerBound) || recordTime.isAfter(upperBound))
-      Left(s"Record time $ledgerTime outside of range [$lowerBound, $upperBound]")
+      Left(s"Record time $recordTime outside of range [$lowerBound, $upperBound]")
```

Comment:
Hm, I think the check is slightly backwards to begin with. `lowerBound` and `upperBound` should be derived from `recordTime`. This determines the window into which `ledgerTime` must fall. The error message should then say something like `Ledger time $ledgerTime outside of range [$lowerBound, $upperBound]"`.

Reply:
You can either check whether `ledgerTime` falls within a window derived from the `recordTime`, or you can check whether `recordTime` falls within a window derived from the `ledgerTime`. The checks are identical, the only difference is the error message to the user.

The check changed in https://github.com/digital-asset/daml/commit/474e92c13c6989fb2ed2b59af58007880818b465#diff-813e8426f7741d927d9f57732c24ba69, should we revert to the previous error message based on `ledgerTime`?